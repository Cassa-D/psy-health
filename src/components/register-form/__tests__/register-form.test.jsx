import React from "react";
import renderer from "react-test-renderer";

import RegisterForm from "../index";

jest.mock("react-hook-form", () => ({
  useForm: () => ({
    register: jest.fn(),
    handleSubmit: jest.fn((func) => func),
    errors: {},
    setError: jest.fn(),
  }),
}));

describe("Tests snapshot", () => {
  it("shows pat", () => {
    const tree = renderer
      .create(
        <RegisterForm
          isPsic={false}
          values={{}}
          onSubmit={() => {}}
          formErrors={{ register: () => {}, handleSubmit: () => {}, errors: {} }}
          handleOnChange={() => {}}
          handleMaskOnChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shows psi", () => {
    const tree = renderer
      .create(
        <RegisterForm
          isPsic
          values={{}}
          onSubmit={() => {}}
          formErrors={{ register: () => {}, handleSubmit: () => {}, errors: {} }}
          handleOnChange={() => {}}
          handleMaskOnChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
