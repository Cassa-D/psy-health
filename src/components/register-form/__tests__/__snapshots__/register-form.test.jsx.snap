// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshot shows pat 1`] = `
<form
  className="sc-bdnylx ljPwdE"
  name="register"
  noValidate={true}
>
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="name"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Nome e sobrenome"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="email"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="E-mail"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="email"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="confirmEmail"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Confirmar E-mail"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="email"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    maxLength={15}
    name="phone"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Telefone (opcional)"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="tel"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    maxLength={18}
    name="cpf_cnpj"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="CPF/CNPJ"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="password"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Senha"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="password"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="confirmPassword"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Confirmar Senha"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="password"
  />
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <button
<<<<<<< HEAD
      className="sc-dlnjPT gwzrKe"
      data-text="Register"
      fontSize="19px"
      height="40px"
      onClick={true}
=======
      className="sc-dlnjPT jJgUMU"
      data-text="Register"
      fontSize="19px"
      height="40px"
      onClick={[Function]}
>>>>>>> submaster
      width="150px"
    >
      <span>
        R
      </span>
      <span>
        e
      </span>
      <span>
        g
      </span>
      <span>
        i
      </span>
      <span>
        s
      </span>
      <span>
        t
      </span>
      <span>
        e
      </span>
      <span>
        r
      </span>
    </button>
  </div>
</form>
`;

exports[`Tests snapshot shows psi 1`] = `
<form
  className="sc-bdnylx ljPwdE"
  name="register"
  noValidate={true}
>
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="name"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Nome e sobrenome"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="email"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="E-mail"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="email"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="confirmEmail"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Confirmar E-mail"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="email"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    maxLength={15}
    name="phone"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Telefone (opcional)"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="tel"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    maxLength={18}
    name="cpf_cnpj"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="CPF/CNPJ"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    maxLength={8}
    name="crp"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="CRP"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="password"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Senha"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="password"
  />
  <input
<<<<<<< HEAD
    className="sc-gtssRu jigLkO"
=======
    className="sc-gtssRu kTOxiL"
>>>>>>> submaster
    name="confirmPassword"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    placeholder="Confirmar Senha"
    style={
      Object {
        "border": "2px solid #70a3ef",
      }
    }
    type="password"
  />
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <button
<<<<<<< HEAD
      className="sc-dlnjPT gwzrKe"
      data-text="Register"
      fontSize="19px"
      height="40px"
      onClick={true}
=======
      className="sc-dlnjPT jJgUMU"
      data-text="Register"
      fontSize="19px"
      height="40px"
      onClick={[Function]}
>>>>>>> submaster
      width="150px"
    >
      <span>
        R
      </span>
      <span>
        e
      </span>
      <span>
        g
      </span>
      <span>
        i
      </span>
      <span>
        s
      </span>
      <span>
        t
      </span>
      <span>
        e
      </span>
      <span>
        r
      </span>
    </button>
  </div>
</form>
`;
